include/rocksdb/memtablerep.h

```diff
@@ -57,6 +57,10 @@ class MemTableRep {
   // concatenated with values.
   class KeyComparator {
    public:
+    typedef rocksdb::Slice DecodedType;
+
+    virtual DecodedType decode_key(const char* key) const = 0;
```

Comment:
I think it's fair to assume the key is a length-prefixed Slice. In MemTable::Add you can see the encoded key format which can't be overridden. Do you think we can provide a default implementation here considering that?

Reply:
The `MemTable::Add` case makes I'm not afraid anymore. I will push reworked commit soon.